libcodechecker/libhandlers/store.py

```diff
@@ -215,71 +206,53 @@ def collect_file_hashes_from_plist(plist_file):
         except Exception as ex:
             LOG.error('Parsing the plist failed: ' + str(ex))
 
-    try:
-        for input_path in inputs:
-            input_path = os.path.abspath(input_path)
+    for input_path in inputs:
+        input_path = os.path.abspath(input_path)
 
-            if not os.path.exists(input_path):
-                raise OSError(errno.ENOENT,
-                              "Input path does not exist", input_path)
+        if not os.path.exists(input_path):
+            raise OSError(errno.ENOENT,
+                          "Input path does not exist", input_path)
 
-            if os.path.isfile(input_path):
-                files = [input_path]
-            else:
-                _, _, files = next(os.walk(input_path), ([], [], []))
+        if os.path.isfile(input_path):
+            files = [input_path]
+        else:
+            _, _, files = next(os.walk(input_path), ([], [], []))
 
+        with zipfile.ZipFile(zip_file, 'a') as zipf:
             for f in files:
                 plist_file = os.path.join(input_path, f)
                 if f.endswith(".plist"):
                     if collect_file_hashes_from_plist(plist_file):
                         LOG.debug(
                             "Copying file '{0}' to ZIP assembly dir..."
                             .format(plist_file))
-                        shutil.copy(plist_file, report_dir)
+                        zipf.write(plist_file, os.path.join('reports', f))
                     else:
                         LOG.warning("Skipping '{0}' because it contains "
                                     "a missing source file."
                                     .format(plist_file))
 
-        necessary_hashes = client.getMissingContentHashes(hash_to_file.keys())
+    necessary_hashes = client.getMissingContentHashes(hash_to_file.keys())
+    with zipfile.ZipFile(zip_file, 'a') as zipf:
```

Comment:
Why do we reopen this zipfile instead of keeping it open?

Reply:
I think it's rather a question of style because it doesn't really cause performance issues. Anyway I can change it.